SparseGrids/tsgLinearSolvers.cpp

```diff
@@ -164,7 +164,26 @@ std::vector<double> TasmanianTridiagonalSolver::getSymmetricEigenvalues(int n, s
     return result;
 }
 
-void TasmanianTridiagonalSolver::decompose(int n, std::vector<double> &d, std::vector<double> &e, std::vector<double> &z){
+void TasmanianTridiagonalSolver::decompose(std::vector<double> &diag, std::vector<double> &off_diag, const double mu0,
+                                           std::vector<double> &nodes, std::vector<double> &weights) {
+    switch(TasmanianTridiagonalSolver::decompose_version) {
+        case 1 :
+            weights.resize(diag.size());
+            nodes.resize(diag.size());
+            nodes = diag;
```

Comment:
Once you use the `=` operator, the nodes will be overwritten completely, no need to resize right before that.

Reply:
Fixed.